modules/distributor/receiver/shim.go

```diff
@@ -118,11 +129,36 @@ func New(receiverCfg map[string]interface{}, pusher BatchPusher, middleware Midd
 		}
 	}
 
-	p := config.NewMapFromStringMap(map[string]interface{}{
-		"receivers": receiverCfg,
+	receivers := make([]string, 0, len(receiverCfg))
+	for k := range receiverCfg {
+		receivers = append(receivers, k)
+	}
+	fmt.Println("receivers", receivers)
+
+	pro, err := service.NewConfigProvider(service.ConfigProviderSettings{
```

Comment:
whoa, this intense. can we get a comment about what all this config does?

Reply:
Added a few comments, but I agree is a bit overwhelming. The process of generating the actual config struct has become cumbersome and requires all that scaffolding.